docs/learn/expert-guides/universal-profile/extend-profile-functionalities.md

```diff
@@ -0,0 +1,556 @@
+
+1. Deploy contracts that have the logic to be added to the Universal Profile
+2. Attach the extension addresses to the Universal Profile
+
+## Introduction
+
+:::info
+
+Read the standard docs about including [LSP17-ContractExtension in the Universal Profile](../../../standards/universal-profile/lsp0-erc725account.md#lsp17---contract-extension).
+
+:::
+
+A Universal Profile contains non-biased features, including generic execution capabilities, the ability to attach generic information, and the ability to be notified and react to various actions. However, post-deployment, a Universal Profile, like any smart contract, becomes immutable. This immutability restricts the ability to add new functionalities into the code of the contract.
+
+To overcome this limitation, the concept of **extensions** is introduced. Extensions are essentially external contracts that act as supplementary modules to the Universal Profile. They provide a dynamic way to enhance the profile's capabilities.
+
+When a call is made to the Universal Profile for a function that isn't natively part of its primary code, the contract checks if this function exists as an extension. If found, the call is seamlessly redirected to the appropriate external contract (extension), thereby effectively expanding the functionality of the Universal Profile without altering its core, immutable code. This approach enables greater flexibility and adaptability, ensuring the Universal Profile can evolve to meet diverse requirements.
+
+## Extending Functionalities
+
+### Setup
+
+Set up a new project with:
+
+```bash
+npm init
+```
+
+Then, install the dependencies
+
+```bash
+npm install ethers@v5 @lukso/lsp-smart-contracts@v0.14
+```
+
+### Step 1: Instantiate the Universal Profile
+
+#### Generate a Signer
+
+First, we need to generate a signer. This signer will be used to interact with the blockchain. We'll use a private key of the Universal Profile controller and an RPC URL.
+
+```js
+const { ethers } = require('ethers');
+
+// RPC URL (e.g., LUKSO testnet)
+const rpcUrl = 'https://rpc.testnet.lukso.network';
+
+// Replace with your private key
+const privateKey = 'your-private-key';
+
+const provider = new ethers.providers.JsonRpcProvider(rpcUrl);
+const signer = new ethers.Wallet(privateKey, provider);
+```
+
+#### Instantiate the Universal Profile Contract
+
+Now, we'll instantiate an instance of the Universal Profile contract.
+
+```js
+const UniversalProfileABI =
+  require('@lukso/lsp-smart-contracts/artifacts/UniversalProfile.json').abi;
+
+// Replace with the address of your Universal Profile
+const universalProfileAddress = 'your-universal-profile-address';
+
+const universalProfile = new ethers.Contract(
+  universalProfileAddress,
+  UniversalProfileABI,
+  signer,
+);
+```
+
+### Step 2: Create the Extension Contract
+
+Next, we will create a Solidity contract named `TipMe` which will inherit from `LSP17Extension` and have a function `tipMe(..)` that emits an event `Tipped`.
```

Comment:
maybe you can link LSP17Extension to

https://docs.lukso.tech/contracts/contracts/LSP17ContractExtension/LSP17Extension

(relative link ofc)